## First ipyrad run

**20170801**

I'm going to use the same parameters I've been working with with scallops to do a first look at the sea cucumber data for comparison.

In [2]:
cd /mnt/hgfs/SHARED_FOLDER/Learn_iPyrad/PARCAL_RUN1

/mnt/hgfs/SHARED_FOLDER/Learn_iPyrad/PARCAL_RUN1


In [4]:
!ipyrad -n parcal_001


 -------------------------------------------------------------
  ipyrad [v.0.6.20]
  Interactive assembly and analysis of RAD-seq data
 -------------------------------------------------------------

  A new version of ipyrad is available (v.0.7.8). To upgrade run:

    conda install -c ipyrad ipyrad


  New file 'params-parcal_001.txt' created in /mnt/hgfs/SHARED_FOLDER/Learn_iPyrad/PARCAL_RUN1



I made a barcodes file:

In [5]:
!head barcodes_parcal1.txt -n 10

1_A7	AAACGG
1_A8	GCCGTA
1_A10	ACTCTT
1_B7	TTCTAG
1_B12	ATTCCG
1_C7	CCGCAT
1_C8	CGAGGC
1_C11	CGCAGA
1_D7	GAGAGA
1_D8	GGGGCG


And I updated the params file:

In [6]:
!head params-parcal_001.txt -n 30

------- ipyrad params file (v.0.6.20)-------------------------------------------
parcal_001                     ## [0] [assembly_name]: Assembly name. Used to name output directories for assembly steps
./                             ## [1] [project_dir]: Project dir (made in curdir if not present)
/mnt/hgfs/Data/PARCAL/PARCAL_RUN1/raw_data/PARCAL_RUN1_1.fq.gz      ## [2] [raw_fastq_path]: Location of raw non-demultiplexed fastq files
./barcodes_parcal1.txt         ## [3] [barcodes_path]: Location of barcodes file
                               ## [4] [sorted_fastq_path]: Location of demultiplexed/sorted fastq files
denovo                         ## [5] [assembly_method]: Assembly method (denovo, reference, denovo+reference, denovo-reference)
                               ## [6] [reference_sequence]: Location of reference sequence file
rad                            ## [7] [datatype]: Datatype (see docs): rad, gbs, ddrad, etc.
TGCAGG,                         ## [8] [restriction_overhan

Then I ran all steps outside the notebook:

```
ipyrad -p params-parcal_001.txt -s 1234567
```

In [4]:
cd parcal_002_outfiles/

/mnt/hgfs/SHARED_FOLDER/Learn_iPyrad/PARCAL_RUN1/parcal_002_outfiles


In [5]:
!head parcal_002_stats.txt -n 15



## The number of loci caught by each filter.
## ipyrad API location: [assembly].stats_dfs.s7_filters

                            total_filters  applied_order  retained_loci
total_prefiltered_loci              52408              0          52408
filtered_by_rm_duplicates            6994           6994          45414
filtered_by_max_indels               5598           5598          39816
filtered_by_max_snps                15585           6579          33237
filtered_by_max_shared_het              0              0          33237
filtered_by_min_sample              30534          25530           7707
filtered_by_max_alleles                 0              0           7707
total_filtered_loci                  7707              0           7707



**20170803**

I took a first look through the stats file, and it looks like one sample had super low read depth, and a few others had lower read depth (maybe 10-20% of the average), so I'll need to toss those out and rerun the assembly. That might give me more loci, as 7707 isn't very much.

I decided to make an arbitrary cut off at 1 million reads, and through out the handful of samples that had less than that for raw reads. Rerunning that branch now, will call if parcal003.

In [8]:
!ipyrad -p params-parcal_002.txt -b parcal_003 parcal1_1M_cutoff.txt


  loading Assembly: parcal_002
  from saved path: /mnt/hgfs/SHARED_FOLDER/Learn_iPyrad/PARCAL_RUN1/parcal_002.json
  creating a new branch called 'parcal_003' with 77 Samples
  writing new params file to params-parcal_003.txt



Now running steps 3-7 of parcal_003 branch, outside the notebook.

### Run was successful!

Here's the top of the stats7 output file:

In [5]:
cd /mnt/hgfs/SHARED_FOLDER/Learn_iPyrad/PARCAL_RUN1/parcal_003_outfiles

/mnt/hgfs/SHARED_FOLDER/Learn_iPyrad/PARCAL_RUN1/parcal_003_outfiles


In [6]:
!head parcal_003_stats.txt -n 15



## The number of loci caught by each filter.
## ipyrad API location: [assembly].stats_dfs.s7_filters

                            total_filters  applied_order  retained_loci
total_prefiltered_loci              51898              0          51898
filtered_by_rm_duplicates            6974           6974          44924
filtered_by_max_indels               5497           5497          39427
filtered_by_max_snps                15418           6537          32890
filtered_by_max_shared_het              0              0          32890
filtered_by_min_sample              30214          25265           7625
filtered_by_max_alleles                 0              0           7625
total_filtered_loci                  7625              0           7625



I did get more loci, but not many more. Probably worth looking into the genome size. Also, I made plots to look for correlation between DNA quality (as verified on a gel before RAD library prep) and read depth and number of loci.

![img](https://github.com/nclowell/SeaCukes/blob/master/Imgs_for_Notebooks/rd_by_DNA_qual_20170804.png?raw=true)


<br>
<br>
<br>

![img](https://github.com/nclowell/SeaCukes/blob/master/Imgs_for_Notebooks/num_loci_by_DNA_qual_20170804.png?raw=true)

So I'd also like to run an ipyrad assembly by tossing out those with poor DNA quality, because although they have pretty high read depth, maybe the sequencing quality is low and many more things are getting tossed out.